test/testrunner.py

```diff
@@ -160,14 +166,23 @@ def main():
         if specific_test_type == "sanity":
             pull_dlc_images(all_image_list)
         if specific_test_type == "eks":
-            new_eks_cluster_name = setup_eks_clusters(dlc_images)
-        # Execute dlc_tests pytest command
-        pytest_cmd = ["-s", "-rA", test_path, f"--junitxml={report}", "-n=auto"]
+            eks_cluster_name = setup_eks_clusters(dlc_images)
+            # We expect image URIs of only one framework for EKS tests.
+            framework = "mxnet" if "mxnet" in dlc_images else "pytorch" if "pytorch" in dlc_images else "tensorflow"
+            # Split training and inference, and run one after the other, to prevent scheduling issues
+            pytest_cmds = [
+                ["-s", "-rA", os.path.join(test_path, framework, "training"), f"--junitxml={report_train}", "-n=auto"],
+                ["-s", "-rA", os.path.join(test_path, framework, "inference"), f"--junitxml={report_infer}", "-n=auto"],
+            ]
+        else:
+            # Execute dlc_tests pytest command
+            pytest_cmds = [["-s", "-rA", test_path, f"--junitxml={report}", "-n=auto"]]
         try:
-            sys.exit(pytest.main(pytest_cmd))
+            cmd_exit_statuses = [pytest.main(pytest_cmd) for pytest_cmd in pytest_cmds]
+            sys.exit(0) if all([status == 0 for status in cmd_exit_statuses]) else sys.exit(1)
         finally:
-            if specific_test_type == "eks":
-                eks_utils.delete_eks_cluster(new_eks_cluster_name)
+            if specific_test_type == "eks" and eks_cluster_name:
```

Comment:
This change is not really necessary, because eks_cluster_name is only assigned/used when `specific_test_type == "eks"`. This change is only to cover all bases, and make debugging easier for developers in the future.